skglm/solvers/anderson_cd.py

```diff
@@ -0,0 +1,345 @@
+import numpy as np
+from numba import njit
+from scipy import sparse
+from sklearn.utils import check_array
+from skglm.solvers.common import construct_grad, construct_grad_sparse, dist_fix_point
+from skglm.solvers.base import BaseSolver
+from skglm.utils import AndersonAcceleration
+
+
+class AndersonCD(BaseSolver):
+    """Coordinate descent solver with working sets and Anderson acceleration.
+
+    fit_intercept : bool
+        Whether or not to fit an intercept.
+
+    max_iter : int, optional
+        The maximum number of iterations (definition of working set and
+        resolution of problem restricted to features in working set).
+
+    max_epochs : int, optional
+        Maximum number of (block) CD epochs on each subproblem.
+
+    p0 : int, optional
+        First working set size.
+
+    tol : float, optional
+        The tolerance for the optimization.
+
+    ws_strategy : ('subdiff'|'fixpoint'), optional
+        The score used to build the working set.
+
+    verbose : bool or int, optional
+        Amount of verbosity. 0/False is silent.
+
+    References
+    ----------
+    .. [1] Bertrand, Q. and Klopfenstein, Q. and Bannier, P.-A. and Gidel, G.
+           and Massias, M.
+           "Beyond L1: Faster and Better Sparse Models with skglm", 2022
+           https://arxiv.org/abs/2204.07826
+
+    .. [2] Bertrand, Q. and Massias, M.
+           "Anderson acceleration of coordinate descent", AISTATS, 2021
+           https://proceedings.mlr.press/v130/bertrand21a.html
+           code: https://github.com/mathurinm/andersoncd
+    """
+
+    def __init__(self, max_iter=50, max_epochs=50_000, p0=10,
+                 tol=1e-4, ws_strategy="subdiff", fit_intercept=True,
+                 warm_start=False, verbose=0):
+        self.max_iter = max_iter
+        self.max_epochs = max_epochs
+        self.p0 = p0
+        self.tol = tol
+        self.ws_strategy = ws_strategy
+        self.fit_intercept = fit_intercept
+        self.warm_start = warm_start
+        self.verbose = verbose
+
+    def solve(self, X, y, datafit, penalty, w_init=None, Xw_init=None):
+        if self.ws_strategy not in ("subdiff", "fixpoint"):
+            raise ValueError(
+                'Unsupported value for self.ws_strategy:', self.ws_strategy)
+        n_samples, n_features = X.shape
+        w = np.zeros(n_features) if w_init is None else w_init
```

Comment:
There is a missing ``self.fit_intercept`` here, right? Though we pass the tests!

Reply:
Can you add a test that makes this break and then fix the faulty code ?

Reply:
It doesn't break when fitting an estimator since we use ``_glm_fit`` function that passes in a ``w_init``.
However, it breaks when using ``AndersonCD`` solver.
I fixed that. On the other hand, I don't think it deserves to be added as a unitest. WDYT @mathurinm?